src/algebra/group/prod.lean

```diff
@@ -482,6 +482,10 @@ def embed_product (α : Type*) [monoid α] : αˣ →* α × αᵐᵒᵖ :=
     and_self],
   map_mul' := λ x y, by simp only [mul_inv_rev, op_mul, units.coe_mul, prod.mk_mul_mk] }
 
+lemma embed_product_injective (α : Type*) [monoid α] : function.injective (embed_product α) :=
+λ a₁ a₂ h, by { rw [embed_product, monoid_hom.coe_mk, prod.mk.inj_iff, op_inj] at h,
+  rw [units.ext_iff, h.1], }
```

Comment:
```suggestion
λ a₁ a₂ h, units.ext $ (congr_arg prod.fst h : _)
```

Reply:
`foo (a b)` is the same as `foo $ a b`

So actually, the `$` is pointless here, I wasn't paying attention!